Nr 1. 
(a) $f(x) = (x³ + \frac{1}{3}) - (x³ - \frac{1}{3})$ 
Wann ist die Abweichung mehr als 1 \%?


In [2]:
import numpy as np

#f = (x**3 + (1/3))-(x**3 - (1/3))
t = (2**3 + (1/3))-(2**3 - (1/3))
x=2

while abs((2/3) - t) < 0.01 :
    x = x * 2
    t = (x**3 + (1/3))-(x**3 - (1/3))
    
    
print(x,t)

65536 0.65625


Beim Ausführen des Codes kommt ein $x = 65.536$ und ein $f = 0.65625$ heraus. D.h. bei ca. $65,5 \cdot 10³$ wird die ABweichung größer als 1 \%.

In [4]:
while np.isclose(t,0) is not True:
    x = x * 10
    t = (x**3 + (1/3))-(x**3 - (1/3))

print(x,t)

OverflowError: int too large to convert to float